src/app/adminDashboard/spectator/page.tsx

```diff
@@ -1,12 +1,331 @@
 
-import React from 'react'
+'use client';
+import React, { useState, useEffect } from "react";
+import styles from "../../../styles/Dashboard.module.scss";
+import { sendRequest } from "@/services/auth/auth_All_Api";
+//@ts-ignore
+import { Input, Button } from 'technogetic-iron-smart-ui';
+import { Navbar } from '../../../Components/Navbar/Navbar';
+import Loader from "@/Components/Loader/Loader";
+import TableData, { SpectatorData } from '../../../Components/Table/Table';
+import { toast } from 'react-toastify';
+import Image from "next/image";
+
+
+interface FormCreate {
+  fullName: string;
+  userName: string;
+  email: string;
+  password: string;
+  role: any;
+}
+
+export default function Modal() {
+  const [spectatorData, setSpectatorData] = useState<SpectatorData[]>([]);
+  console.log("spectatorData", spectatorData)
+  const [isLoading, setIsLoading] = useState(true);
+  const [isupdateData, setIsUpdateData] = useState<any>(null)
+  const imageIcon: string = 'spectator';
+  const [getSpectatorId, setSetSpectatorId] = useState<any>()
+
+
+  const [modal, setModal] = useState(false);
+  const [formErrors, setFormErrors] = useState<FormCreate>({
+    fullName: '',
+    userName: '',
+    email: '',
+    password: '',
+    role: '64d7239c8a677a5d2e21b00d'
+  });
+
+  const getAllUsers = async () => {
+    try {
+      const tokens = localStorage.getItem('jwtToken');
+      const allUsersData: any = await sendRequest('/user/getalluser', {
+        method: 'GET',
+        headers: { 'Authorization': `Bearer ${tokens}` }
+      });
+
+      const allspectatorData = allUsersData?.data?.data;
+      const filteredData = allspectatorData.filter((spectator: any) => {
+        return spectator.role.role === 'spectator';
+      });
+
+      setSpectatorData(filteredData);
+      setIsLoading(false);
+    } catch (error) {
+
+      setIsLoading(false);
+    }
+  };
+
+
+  const deleteroomId = async (userUuid: any) => {
+    try {
+      const tokens = localStorage.getItem('jwtToken');
+      console.log(tokens)
+      const response: any = await sendRequest(`/role/deleterole/${userUuid}`, {
+        method: 'delete',
+        headers: { 'Authorization': `Bearer ${tokens}` }
+      });
+      getAllUsers();
+      if (response) {
+        const success = response.data.message;
+        toast.success(success);
+      }
+      setSpectatorData(spectatorData);
+    } catch (error) {
+      console.error('Error deleting room:', error);
+
```

Comment:
should have toast error as well and loading state to be false in case of success and error

Reply:
done sir